src/main.cpp

```diff
@@ -219,6 +220,22 @@ namespace
 
 				bool moved_this_frame = false;
 				vec3 move_dir{};
+				if (win.key('+', 61))
+				{
+					current_renderer_index = (current_renderer_index + 1) % renderers::all().size();
+					const auto& all_renderers = renderers::all();
+					regular_renderer = create_renderer(all_renderers[current_renderer_index].name);
+					reload_requested = true;
+				}
+
+				if (win.key('-', 45))
+				{
+					current_renderer_index = (current_renderer_index - 1) % renderers::all().size();
+					const auto& all_renderers = renderers::all();
+					regular_renderer = create_renderer(all_renderers[current_renderer_index].name);
+					reload_requested = true;
+				}
```

Comment:
These operations are the same -  you could combine these two branches into one and change `current_renderer_index + 1` to `current_renderer_index + <plus or minus one depending on which key>`

Keep things [DRY](https://docs.getdbt.com/terms/dry) :)

Reply:
Oh, also, this:

```cpp
const auto& all_renderers = renderers::all();
```

should be this:

```
auto all_renderers = renderers::all();
```

renderers::all() does not return a reference - the only reason what you have even worked was because you made it a `const` reference and were trigging a weird special case called [lifetime extension](https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/). It's "legal", but generally you should avoid it in non-template code.

Reply:
OH also! you don't need to set `reload_requested = true;`. That's for the live reloading of the scene. You're changing the renderer here. Totally different thing.

Reply:
If I don't live reload the scene, how will should I update the rendered that is being used?

Reply:
That's already taken care of just by virtue of you updating the `regular_renderer` variable
 Changing it should "just work" without having anything to do with the scene.

Reply:
That is, the current renderer and the current scene are two different abstractions; changing one does not require doing anything to the other. They should be sufficiently de-coupled.